Documentation/sources/push-notification/automatic-handling-and-tracking.md

```diff
@@ -0,0 +1,85 @@
+# Automatically display and track push notification
+
+## Overview
+This document describes how to automatically display and track push notifications using the AEPMessaging extension. We highly recommend to use this method unless you have a specific requirement to manually display and track push notifications.
+
+## Sync the push token
+
+To register the push token with Adobe Experience Platform, add the following code to the Application class's `onCreate()` method:
+
+```java
+public class YourApp extends Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        FirebaseMessaging.getInstance().getToken().addOnCompleteListener(new OnCompleteListener<String>() {
+
+            @Override
+            public void onComplete(@NonNull Task<String> task) {
+                if (task.isSuccessful()) {
+                    String token = task.getResult();
+                    MobileCore.setPushIdentifier(token);
+                }       
+            }
+        });
+    }
+}
+```
+
+## Display and tracking push notification
+
+#### Register Messaging extension's FirebaseMessagingService
+
+> **_NOTE:_** 
+ If your application already registered a FirebaseMessagingService class, then you can skip this step and move to [Using your own FirebaseMessagingService](#using-your-own-firebasemessagingservice) section.
```

Comment:
and move to **the** [Using your own FirebaseMessagingService](#using-your-own-firebasemessagingservice) section.